storage/s3.go

```diff
@@ -302,12 +304,15 @@ func s3Config(cfg S3Config) aws.Config {
 		Credentials: credentials,
 	}
 	if cfg.Endpoint != "" {
-		config.EndpointResolverWithOptions = aws.EndpointResolverWithOptionsFunc(func(service, region string, options ...interface{}) (aws.Endpoint, error) {
-			return aws.Endpoint{
-				URL:           "https://" + cfg.Endpoint,
-				SigningRegion: cfg.Region,
-			}, nil
-		})
+		if endpointIsAllowed(cfg.Endpoint) {
```

Comment:
Suggestion(wording):
```suggestion
		if isEndpointAllowed(cfg.Endpoint) {
```